### DISCUSSION TITLE
Option hotkeys in obsidian on a mac

### DISCUSSION THREAD

**Discussion by diego898** [BryanJenksCommunity/FAQ#94] [Q&A] [answered]
hello, this is a bit silly, but in my mac when I press `option+anychar`, I get special characters in all apps. So, I can't have shortcuts in obsidian like `option+r`. How do you have that working on your mac?

I am watching your spaced repetition video and I noticed you use these hotkets.

**Comment by tallguyjenks** [comment, marked as answer]
When mapping hot keys you can use option just fine. It's what I do. Maybe they've fixed the display so it shows "alt" instead of option so it's interoperable with windows but there should be no issues :)

**Reply by diego898** [reply]
Right. When making hotkeys, I can use option to set a hotkey. But while in in the app (like the editor window), If I then try to use the hotkey, it does not work. Instead, I see a special character.

**Reply by tallguyjenks** [reply]
mmmm thats strange. because mine do not do that. I have option+P to toggle preview mode, it doesn't type a π symbol in obsidian.

**Reply by diego898** [reply]
yeah when I assign a new hotkey I can access it. For example:

<img width="232" alt="image" src="https://user-images.githubusercontent.com/742407/127065509-f9e53165-bbae-42e5-83b6-c78b2d696e9b.png">

You can see the arrow with a dot to revert it, meaning Ive successfully assigned it to the option+n hotkey. However, when I actually type that hotkey I see the following: `˜`, which is distinct from the normal `~` character.

This happens in any app, not just obsidian, and on my mac keyboard or USB keyboard. 

When I check keyboard in control panel, I see:

<img width="656" alt="image" src="https://user-images.githubusercontent.com/742407/127065692-18599eca-2b1d-4edd-8e72-ee3aac06dc81.png">

And on that screen, when I press and hold down option, I see the  keyboard change to show all "special" characters.

I'm sorry to spring this on you, but I cant find anything through regular searching, and asking on the discord didnt help. I thought this just might be a mac thing, but I saw you can actually use the key combo as intended on mac. 

Any help you can provide would be greatly appreciated!

**Reply by tallguyjenks** [reply]
yeah i see the exact same thing you do. i don't see any settings to change anything regarding this behavior on the mac side of things. don't know about this one

**Reply by datavistics** [reply]
I have the same issue...

**Reply by yarub123** [reply]
@diego898 Did you ever figure this out? 😞

**Reply by adamallo** [reply]
I am also having precisely the same issue. I wonder if this is related to having the modifier keys remapped for my USB keyboard (option <--> command) (System settings --> Keyboard --> Keyboard shortcuts --> Modifier Keys).
@diego898, @yarub123, @datavistics, Do you guys also have the modifier keys remapped?

**Reply by datavistics** [reply]
I did have the modifier keys remapped. But I dont _think_ this is the issue for 2 reasons. 
1. It doesnt work when I use the mac keys directly.
2. The input source shows what happens when you hit modifier keys, and I think its related to that.

**Reply by yarub123** [reply]
@datavistics @adamallo  oops forgot about this post, I figured it out. Just change your keyboard input in the mac settings. "Unicode Hex Layout" is one that works.

- You know how you add another language to your keyboard? It's that same method, but when adding this Unicode Layout, in the keyboards section go down to "Other" all the way on the bottom of the list, once there, select Unicode Hex Layout and add it to your list of keyboards.

- I used this method, it works, I also used this other method and it also works. Go ahead and download this layout titled "x_layout"..... it is all the letters on your QWERTY keyboard but without the accent symbols etc. 

-Edit: [here's the link](https://gist.github.com/haosdent/573ea124e5ea666fc576). Download this layout, and add it as if you're adding another language in Keyboard settings on Mac.

- After using these two methods you can simply switch keyboards from the menubar.

Let me know if you guys have any questions. 
Peace be with you.

**Reply by datavistics** [reply]
@yarub123 
Ive tried the hex layout and it didnt work for me. Here are my settings:
![image](https://user-images.githubusercontent.com/22736772/225526559-ea798cfc-a71a-4f7b-9b41-583f9ba48f65.png)
![image](https://user-images.githubusercontent.com/22736772/225526597-a52951d9-585c-46a3-9235-a76fa6db8b53.png)
Is it the same for you?

**Reply by yarub123** [reply]
That's fine. Go ahead and try the link I posted in previous comment. The layout in the link I sent yesterday should solve it for you. Let me know if it works.

**Reply by datavistics** [reply]
Hrm... Im not sure what Im doing wrong. I installed it. Here are the [instructions](https://languagegeek.com/lgwp/keyboards/installing-mac-keyboard-layouts/) for anyone wondering.

Its still not working. 

@yarub123 , what do you see when you click on 
<img width="212" alt="image" src="https://user-images.githubusercontent.com/22736772/225831322-173e3deb-edf7-4576-ada6-27b82c258ab7.png">
And press option? Are they mostly blank?

**Reply by datavistics** [reply]
I think they should show the original keys, and mine are blank. Im wondering if that is the issue.

**Reply by yarub123** [reply]
@datavistics No that's fine. Now go ahead and try to use whatever hotkey you wanted in obsidian. It will now work instead of showing a weird character. 

For example if I press "⌥+2" it will show " ™ "
- but with "x_layout" selected, it will NOT show the above. 
- You can now use this in the Obsidian for your hotkeys. 

- And yes even Unicode Hex Input will show up as blank when you hold "⌥". That is what it's supposed to do 🙃

**Reply by datavistics** [reply]
Well Im a bit stumped. 
I tried it again, it didnt work. 
I reset the key to "⌥+E" It didnt work. 
I reset the key to "⌥+S" and tried it, and it DID work. 
Re set the key back to "⌥+E" and it worked. 
It also works on the Unicode Hex Input. 

In any case, thank you SO much for helping. Im really not sure what worked. But Im really happy it does, and Im really grateful for your responses.

🤷‍♂️

**Reply by yarub123** [reply]
Haha no worries, I'm happy that it helped! 

I think you were just confusing yourself, maybe also because the blank keys threw you off hehe. All good. Enjoy :)

**Comment by piepak3** [comment]
Just wanted to comment that I had the same issue. What fixed it was just sticking with the default keyboard and resetting the shortcut. Still using "⌥+R" (in this case I am using templater) but just resetting it from the default.  You can see in the screenshot that it isn't the default but the characters are the exact same.

<img width="801" alt="image" src="https://github.com/BryanJenksCommunity/FAQ/assets/58745786/1618fece-a317-4307-b19c-6a258222668b">
